server/util/claims/claims.go

```diff
@@ -202,7 +202,7 @@ func ClaimsFromSubID(ctx context.Context, env environment.Env, subID string) (*C
 
 			// If the user requested impersonation but the subdomain doesn't
 			// match the impersonation target then don't enable impersonation.
-			if sd := subdomain.Get(ctx); sd != "" && ig.URLIdentifier != nil && sd != *ig.URLIdentifier {
+			if sd := subdomain.Get(ctx); sd != "" && sd != ig.URLIdentifier {
```

Comment:
does this need to check `ig.URLIdentifier != ""`?

Reply:
No. I think the old code was incorrect. If the subdomain is set, we should ignore the impersonation if the url identifier doesn't match regardless of whether the identifier is set or not set.